app/api/auth/agent-connect/callback/route.ts

```diff
@@ -25,12 +34,49 @@ export const GET = withSession(async function callbackRoute(req) {
     }
   } catch (e: any) {
     logFatalErrorInSentry(new AgentConnectFailedException({ cause: e }));
-    if (e instanceof HttpForbiddenError) {
+    if (e instanceof AgentConnectCouldBeAServicePublicException) {
+      return NextResponse.redirect(
+        getBaseUrl() + '/connexion/habilitation/requise'
+      );
+    } else if (e instanceof HttpForbiddenError) {
       return NextResponse.redirect(
-        getBaseUrl() + '/connexion/echec-authorisation-requise'
+        getBaseUrl() + '/connexion/habilitation/refuse'
       );
     } else {
       return NextResponse.redirect(getBaseUrl() + '/connexion/echec-connexion');
     }
   }
 });
+
```

Comment:
```suggestion
/**
* Checks if user is either an agent or has habilitation
* Otherwise raise an exception : HttpForbiddenError or AgentConnectCouldBeAServicePublicException
*/
```